interval/search.go

```diff
@@ -426,3 +426,37 @@ func (st *MultiValueSearchTree[V, T]) Select(k int) ([]V, bool) {
 
 	return interval.vals, true
 }
+
+// End returns the values in the tree that have the largest ending interval.
+func (st *MultiValueSearchTree[V, T]) MaxEnd() ([]V, bool) {
+	st.mu.Lock()
+	defer st.mu.Unlock()
+
+	var vals []V
+	if st.root == nil {
+		return vals, false
+	}
+
+	maxEnd(st.root, st.root.maxEnd, st.cmp, func(n *node[V, T]) {
+		vals = append(vals, n.interval.vals...)
+	})
+	return vals, true
+}
+
+func maxEnd[V, T any](n *node[V, T], searchEnd T, cmp CmpFunc[T], visit func(n *node[V, T])) {
```

Comment:
On the definition of the `visit` parameter, the `*node[V, T]` seems pretty descriptive. What do you think of removing the `n` from that definition?
```suggestion
func maxEnd[V, T any](n *node[V, T], searchEnd T, cmp CmpFunc[T], visit func(*node[V, T])) {
```